New CI Tutorial: Build and Push to AWS ECR (#250)
harness/developer-hub: tutorials/build-code/ci-build-push-to-ecr.md

![build pipeline](./static/ci-tutorial-build-push-ecr/configure_pipeline.png)

Once you click on ‘Create Pipeline’, you should see the skeleton of your CI pipeline.
![create pipeline](./static/ci-tutorial-build-push-ecr/build_node_app.png)

You can click on the name ‘Build NodeJS’ and modify the name accordingly. Click on the ‘Execution’ tab and then on ‘Build Node App’. You should see the ‘Run’ step configured for you automatically. 
![run step configure](./static/ci-tutorial-build-push-ecr/run_step_configuration.png)

Now, you can modify the commands. Since we don’t want the first three lines in this project, we will remove them and just keep the ‘npm test’ command. Apply changes and save the pipeline.

The pipeline is ready till testing and building the application. What is left is to push our built image to Amazon ECR. So, let us connect our AWS account with Harness to make sure they communicate with each other. 

In the project setup, go to the ‘Connectors’ tab to connect our AWS account.

![explore connectors](./static/ci-tutorial-build-push-ecr/explore_connectors.png)

![connectors tab](./static/ci-tutorial-build-push-ecr/connectors.png)

> Select AWS from the list and add the required details. 

Reviewer: ```suggestion
9. Select AWS from the list and add the required details. 
```

![select AWS](./static/ci-tutorial-build-push-ecr/aws_connector_overview.png)

> There are three ways to connect your AWS account. We will use ‘AWS Access Key’ method to authenticate and connect.

Reviewer: ```suggestion
10. There are three ways to connect your AWS account. We will use **AWS Access Key** method to authenticate and connect.
```

![AWS access](./static/ci-tutorial-build-push-ecr/AWS_Access.png)

Then we will connect the Harness platform as our option to connect with AWS.
![connect AWS](./static/ci-tutorial-build-push-ecr/connect_provider.png)

> Once you click ‘Save and Continue’, you should see a successful connection message.

Reviewer: ```suggestion
After you select **Save and Continue**, you should see a successful connection message.
```

![cloud provider](./static/ci-tutorial-build-push-ecr/cloud_provider_success.png)

Now, you have successfully connected your AWS account with Harness.

> It is time we need to add our last step in the CI pipeline, i,e pushing the image to Amazon ECR.

Reviewer: ```suggestion
It is time to add our last step in the CI pipeline; pushing the image to Amazon ECR.
```

> Go back to your pipeline and add a step under the ‘Build and Test’ stage (under execution). 

Reviewer: ```suggestion
12. Go back to your pipeline and add a step under the **Build and Test** stage (under execution). 
```

![step library](./static/ci-tutorial-build-push-ecr/step_library.png)

> Select ‘Build and Push to ECR’ 

Reviewer: ```suggestion
13. Select **Build and Push to ECR**.
```

![build and push setup](./static/ci-tutorial-build-push-ecr/build_push_step.png)

> Make sure to correctly add the Region, Account ID and Image Name. Add the ‘Tags’ [I have added ‘testing’ as a Tag]. Apply changes and save the pipeline settings.

Reviewer: ```suggestion
14. Make sure to correctly add the Region, Account ID, and Image Name. Add the Tags [I have added **testing** as a Tag]. Apply the changes and save the pipeline settings.
```

Finally, your pipeline should look like this.
![ecr tutorial](./static/ci-tutorial-build-push-ecr/ECR_tutorial.png)

Basically, we are testing the application with a simple ‘npm test’ command as configured in the ‘Run’ step and pushing the built image to Amazon ECR (configured in the last step).

> Save and run the pipeline:)

Reviewer: ```suggestion
Save and run the pipeline.
```
